### DISCUSSION TITLE
Please, How to select Date picker, execute the Rest Querry and display Row of data according to date selected ?

### DISCUSSION THREAD

**Discussion by assia-zoubir** [Budibase/budibase#13632] [Help] [answered]
please in budibase, how can i select a date in date picker, execute RestQuerry (get) and display the rows of data in to the screen. thank you

**Comment by mikesealey** [comment, marked as answer]
Hey @assia-zoubir 

Here's a little step-by-step guide for you that should get you started, but as always you should adjust to better suit your needs once you're up and running.

In the `Data` tab, you should create your REST query, there's a great section in the documentation which you can find [here](https://docs.budibase.com/docs/rest), but if you're in a hurry, head for the subsection [REST bindings (tutorial)](https://docs.budibase.com/docs/rest-bindings).

Then in the `Design` section, add a `Form` component, and then add a `Date Picker`. Set the `Date Picker` `Field` value to something by typing or selecting from your schema if you have one. Then using the `On Change` set two actions: 
* `Execute Query` - _This should be the REST API query you created in the Data tab._
* `Update State` - _Set State with a key of "response" and a value that is the response from your query. In my case it looks like `{{ Action1.Query result }}` but yours may be different._ 

You then have access to the response of the query, held in state. If you wanted to save these values into the Data screen, you should have a table set up that matches the keys of the JSON response you'll get. You could use the default-value feature of form fields to collect the values of the response using a binding, and save the response that way. You will need to prop-drill into the response body for each specific value. Lastly you'll need to save the information to the table. Additional reading: [Forms Documentation](https://docs.budibase.com/docs/forms)

I always find a Headline component set the value of the response is a useful tool in debugging while you get your functionality up and running, but don't forget to remove it before the user sees your work.

Hopefully that gets you on your way. If you haven't already, you should check out our Discord server, where we have a strong community of users. [Discord Invite Link](https://discord.gg/ZepTmGbtfF)
